File: app/src/main/java/com/my/televip/base/BaseMethodHook.java
package com.my.televip.base;

import com.my.televip.logging.Logger;

import de.robv.android.xposed.XC_MethodHook;

public abstract class BaseMethodHook extends XC_MethodHook {

    protected void beforeMethod(MethodHookParam param) throws Throwable {}

    protected void afterMethod(MethodHookParam param) throws Throwable {}


    @Override
    protected final void beforeHookedMethod(MethodHookParam param) {
        try {
            beforeMethod(param);
        } catch (Throwable e) {
            onError(e);
        }
    }


    @Override
    protected final void afterHookedMethod(MethodHookParam param) {
        try {
            afterMethod(param);
        } catch (Throwable e) {
            onError(e);
        }
    }


    protected void onError(Throwable throwable) {
        Logger.e(throwable);
    }
}

File: app/src/main/java/com/my/televip/base/BaseMethodHook.kt
package com.my.televip.base

import com.my.televip.logging.Logger
import de.robv.android.xposed.XC_MethodHook

abstract class BaseMethodHook : XC_MethodHook() {

    @Throws(Throwable::class)
    protected open fun beforeMethod(param: MethodHookParam) {}

    @Throws(Throwable::class)
    protected open fun afterMethod(param: MethodHookParam) {}

    override fun beforeHookedMethod(param: MethodHookParam) {
        try {
            beforeMethod(param)
        } catch (e: Throwable) {
            onError(e)
        }
    }

    override fun afterHookedMethod(param: MethodHookParam) {
        try {
            afterMethod(param)
        } catch (e: Throwable) {
            onError(e)
        }
    }

    protected open fun onError(throwable: Throwable) {
        Logger.e(throwable)
    }
}
